services/wallet/transfer/transaction.go

```diff
@@ -252,11 +260,30 @@ func (tm *TransactionManager) AddPending(transaction PendingTransaction) error {
 		transaction.AdditionalData,
 		transaction.MultiTransactionID,
 	)
+
+	// Notify listeners of new pending transaction (used in activity history)
+	if err == nil {
+		tm.notifyPendingTransactionListeners(transaction.ChainID, []common.Address{transaction.From, transaction.To}, transaction.Timestamp)
+	}
 	return err
 }
 
+func (tm *TransactionManager) notifyPendingTransactionListeners(chainID uint64, addresses []common.Address, timestamp uint64) {
+	if tm.eventFeed != nil {
+		tm.eventFeed.Send(walletevent.Event{
+			Type:     EventPendingTransactionUpdate,
+			ChainID:  chainID,
+			Accounts: addresses,
+			At:       int64(timestamp),
+		})
+	}
+}
+
 func (tm *TransactionManager) DeletePending(chainID uint64, hash common.Hash) error {
 	_, err := tm.db.Exec(`DELETE FROM pending_transactions WHERE network_id = ? AND hash = ?`, chainID, hash)
+	if err == nil {
+		tm.notifyPendingTransactionListeners(chainID, nil, 0)
```

Comment:
shouldn't we get the from/to addresses from the deleted transaction to put them in the notification?

Reply:
I intentionally left out this optimization given that this should be temporary and improve with the implementation of dynamic updates in the future.

Reply:
I followed your advice in the end; thanks for the heads up. But also found that the to_address in pending is wrong. I will follow up on that issue.